File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.py
import sys

s = set()
for _ in range(int(sys.stdin.readline())):
    name, log = sys.stdin.readline().split()
    if log == "enter":
        s.add(name)
    elif log == "leave":
        s.remove(name)

for i in sorted(s, reverse=True):
    print(i)

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.java
import java.util.*;
import java.io.*;
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        Set<String> logSet = new TreeSet<>(Collections.reverseOrder());
        String name = "";
        String log = "";


        for(int i = 0; i<n; i++){
            st = new StringTokenizer(br.readLine());
            name = st.nextToken();
            log = st.nextToken();

            if(!logSet.contains(name)) {
                if (log.equals("enter")) logSet.add(name);
            } else {
                if (log.equals("leave")) logSet.remove(name);
            }
        }
        // Iterator 사용
        for (String s : logSet) {
            System.out.println(s);
        }


    }
}
